contracts/Lock.sol

```diff
@@ -117,25 +123,52 @@ contract Lock is AragonApp, IForwarder, IForwarderFee {
     /**
     * @notice Locks the required amount of tokens and executes the specified action
     * @dev IForwarder interface conformance. Consider using pretransaction on UI for necessary approval.
+    *      We check for edge cases where an approve is made and before this function is called, one of msg.sender current locks is unlocked resulting in a different griefing amount.
     * @param _evmCallScript Script to execute
     */
-    function forward(bytes _evmCallScript) public {
+    function forward(bytes _evmCallScript) public  auth(LOCK_TOKENS_ROLE) {
```

Comment:
The `auth(LOCK_TOKENS_ROLE)` needs to be called manually in `canForward()` so the UI can determine whether the address can call `forward`. Eg
```
function canForward(address _sender, bytes) public view returns (bool) {
    // Note that `canPerform()` implicitly does an initialization check itself
    return canPerform(_sender, LOCK_TOKENS_ROLE, arr());
}
```
You can then remove the `auth(LOCK_TOKENS_ROLE)` modifier above.

Also remove the currently commented lines from `canForward()`. As a rule I don't believe there's any need to keep unused code around, including the commented function now at the end of the `redemptions` contract, as we have source control and can check out earlier versions if we need to see it again.